### DISCUSSION TITLE
BRouter or some of all the other Services

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#169] [Q&A] [answered]
Yesterday I had to support a friend in setting up Cruiser. He took out a subscription (because the one from K2 had just expired) and from today wanted to be relatively offroad in the Alps with some friends for few days. Panic! 
I helped him install BRouter (documentation from the Kurviger blog - okay, I could have taken the "BRouter guide" from this forum as well) and explained the most important things. Whereby the most difficult discussion was: why BRouter and not one of the many other services? 
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/d431706c-20af-4a5e-adda-54f79590443a"/>

I made it short: I only know BRouter, plus it's guaranteed to be usable offline - period!
Maybe we should "hide" the multitude of services in a subitem for advanced users?
With "BRouter: via28-position not mapped in existing datafile" could fortunately also help - that had been explained to me some months ago here in the forum. I hope he now has a great tour in the nowhere of the Alps.

**Comment by devemux86** [comment, marked as answer]
@WolfgangZ42 thanks for your support!

I like to think that **_with Cruiser you can literally customize everything and make the app your own_**!

(instead of running apps with few options and only as their developers want them to run)

Τoo many options at first sight might be difficult, there is the ongoing process of organizing them better.
All your ideas are welcome!

---

About the routing services, we could split them into 2 groups: Offline and Online?

The "not mapped" message is coming from BRouter when there is no downloaded data for the location.
I can see if I can intercept it and display a more meaningful message?

**Reply by WolfgangZ42** [reply]
> The "not mapped" message is coming from BRouter when there is no downloaded data for the location.

If a waypoint is far away of any path you get "not mapped" as well:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/81403c7c-e0cd-4784-b9dc-ce1c773da29d"/>

**Reply by devemux86** [reply]
@WolfgangZ42 you are right!

This message is when waypoints cannot be attached to the nearest road because they are far away.
The message when trying to do route planning in areas without downloaded data is different.

**Reply by devemux86** [reply]
BRouter is more strict with snap of waypoints on road network than other routers.

Is there a setting in the BRouter profiles that can change this?

**Reply by 0709wiwiwi** [reply]
With Brouter, the snap range from a waypoint to a matching road is limited to **250m**.  (= **corrected** updated)
So you should move waypoints nearer to a **matching** road this according to the set **profile** 

I also don't think the snap range should be too far because then the routing result would also end up being confusing.

What is missing is so a clear popup message.
Who formulates a short fine popup msg ?

**Reply by devemux86** [reply]
> With Brouter, the snap range from a waypoint to a matching road is limited to 150m.

@0709wiwiwi

Where is it? Can it be changed?

>  Who formulates a short fine popup msg ?

This message is from BRouter, maybe I can check it and display something else similar.

**Reply by 0709wiwiwi** [reply]
> Where is it? Can it be changed?

No idea. Alex communicated once.

**Comment by Ciutnik** [comment]
I also wanted to propose (in the future) to separate **all Cruiser options** into settings available for "beginner users" with a basic set of settings, and a whole set of possible settings for "advanced  users", aware of the huge possibilities of personalization. 

I have encountered this solution several times in various applications. The only problem may be recognizing what is "basic" and what is "advanced" - it can be different for everyone 🙂

**Reply by devemux86** [reply]
Most settings categories already have an advanced group.
We could probably organize them differently?

Some groups, which are advanced, such as navigation distances or speeds,
they are still on the first level below their parent, to avoid clicking multiple times to find the settings.

> I have encountered this solution several times in various applications.

Can you mention some apps?
It would be interesting to see how they organize their settings.

**Reply by Ciutnik** [reply]
> Can you mention some apps?

Switching the checkbox causes the appearance or disappearance of some settings that are not crucial for the operation of the application, but useful for advanced users. And so, e.g.:

_Notify for Amazfit_
<img src="https://github.com/devemux86/cruiser/assets/122647782/e6e246d7-1dda-47b6-820b-fcbd82cca7d3" width="33%"/>

_Tasker (this setting seems to have been deprecated due to the fact that Tasker is intended for advanced users)_
<img src="https://github.com/devemux86/cruiser/assets/122647782/ed5ac3eb-3726-415a-b74c-364f876945e2" width="33%"/>

_Ultra GPS Logger_ (it's probably an exaggeration)
<img src="https://github.com/devemux86/cruiser/assets/122647782/4d2db2f9-0a92-4064-872b-a04d4e07b7f4" width="33%"/>

**Reply by devemux86** [reply]
@Ciutnik thanks!
The 1st example seems interesting.

There is already a switch for normal / advanced menu in settings.
We could include an option for advanced settings visibility as well.

**Comment by 0709wiwiwi** [comment]
Locus has a button expert setting. 
It also does have a lot of appeal. 
Who doesn't want to be an expert ?
;-)

**Reply by Ciutnik** [reply]
> Who doesn't want to be an expert ?

I can't imagine it! 😃 I found the above screenshot on the internet, because I couldn't find it in my Tasker... But they were going off topic!
